### PR TITLE
 nice to have : add empty value check

### PR DESCRIPTION
**Pull request by liquanchen9**
nice to have : add empty value check

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@liquanchen9 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23082)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23082) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@liquanchen9 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23082)!

**Review by ielatif — CHANGES_REQUESTED** [review]
You can use `org.springframework.util.StringUtils#hasLength()`

File: spring-web/src/main/java/org/springframework/http/server/ServletServerHttpRequest.java [hunk 1]
```diff
@@ -249,7 +249,7 @@ private static InputStream getBodyFromServletRequestParameters(HttpServletReques
 			for (Iterator<String> valueIterator = values.iterator(); valueIterator.hasNext();) {
 				String value = valueIterator.next();
 				writer.write(URLEncoder.encode(name, FORM_CHARSET.name()));
-				if (value != null) {
+				if (value != null && value.length() > 0) {
```
Comment (ielatif) [inline spring-web/src/main/java/org/springframework/http/server/ServletServerHttpRequest.java]: You can use org.springframework.util.StringUtils#hasLength()

**Review by liquanchen9 — COMMENTED** [review]
ok

**Comment by rstoyanchev** [comment]
A query parameter with an empty value is actually possible.

**Pull request closed without merging**